cal/schema.py

```diff
@@ -36,6 +38,40 @@ def to_json(self):
             "user_id": self.user_id
         }
 
+    @staticmethod
+    def fuzzy_contains(event):
+        """Checks if an event exists in the database.
+
+        Searches by name and date range in case name or date is slightly different.
+
+        :event Event - The event being checked
+        :return Boolean - True if event exists in database, False otherwise
+        """
+        time_fuzz = dt.timedelta(days=3)
+
+        query = (Event.start < event.start + time_fuzz) & \
+                (Event.start > event.start - time_fuzz)
+        if event.end is not None:
+            query &= (Event.end is None) | \
+                ((Event.end < event.end + time_fuzz) &
+                 (Event.end > event.end + time_fuzz))
+
+        # Search for good date matches.
+        date_matches = Event.query.filter(query).all()
+
+        # Filter for good name matches.
+        res = [e for e in date_matches
+               if fuzz.partial_ratio(e.name, event.name) >= 70]
```

Comment:
Could you document that this is an arbitrary cutoff that needs tweaking? Maybe a `# TODO`?